lib/puppet/type/file/content.rb

```diff
@@ -119,6 +107,21 @@ def insync?(is)
       result
     end
 
+    def property_matches?(current, desired)
+      basic = super
+      # The inherited equality is always accepted, so use it if valid.
+      time_types = ["mtime", "ctime"]
+      checksum_type = resource.parameter(:checksum).value.to_s
+      return basic if basic || !time_types.include?(checksum_type)
```

Comment:
nit, can we just compare symbols instead of converting to strings to do the comparison? In ruby, all occurrences of a symbol refer to the same instance, so it doesn't generate new objects.